src/praisonai-agents/praisonaiagents/context/optimizer.py

```
@@ -576,6 +576,128 @@ def _llm_summarize(self, messages: List[Dict[str, Any]], max_tokens: int) -> str
+                if savings_ratio >= self.min_compaction_ratio:
+                    return result, OptimizationResult(
+                        original_tokens=original_tokens,
+                        optimized_tokens=optimized_tokens,
+                        tokens_saved=tokens_saved,
+                        strategy_used=OptimizerStrategy.CONVERSATION,
+                        messages_removed=context.original_message_count - context.compacted_message_count,
+                        summary_added=True,
+                    )
+            except Exception:
+                # Fall back to summarization if conversation compaction fails
+                pass
```

Ruff (0.15.15)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)